### PR TITLE
Adding Android CA Store to the default list of trust managers.

### PR DESCRIPTION
**Pull request by mpavelka**
This PR makes SeaCat use the default Android CA store for validation of peer certificates and allows developers to provide their own trust managers with own trusted certificates.

It also exposes array of `trustManagers` so that it can be used for example in combination with OKHTTP

```
OkHttpClient.Builder()
    .sslSocketFactory(
        seacat.sslContext.socketFactory,
        seacat.trustManagers[0] as X509TrustManager
    )
    .build()
```

### CODE REVIEW

File: seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt [hunk 1]
```diff
@@ -4,17 +4,19 @@ import android.content.Context
...
 import java.util.concurrent.Callable
 import java.util.concurrent.Executors
 import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import javax.net.ssl.TrustManagerFactory
 import javax.net.ssl.X509KeyManager
-import javax.net.ssl.X509TrustManager
 
-class SeaCat(
+open class SeaCat(
```
Comment (ateska) [inline seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt:19]: Needed?

File: seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt [hunk 2]
```diff
@@ -96,25 +103,32 @@ class SeaCat(
...
+     *
+     *  By default there is only one trust manager
+     *  returned that is initialized with the default
+     *  Android CA store.
+     *
+     *  You can override this method and return your own
+     *  array of [TrustManager]s to provide trust to certificates
+     *  issued by your company or for certificate pinning.
+     */
+    protected fun onCreateTrustManagers() : Array<TrustManager> {
```
Comment (ateska) [inline seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt:124]: Should be called `createTrustManagers()`

File: seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt [hunk 3]
```diff
@@ -96,25 +103,32 @@ class SeaCat(
...
-                    override fun checkClientTrusted(chain: Array<out X509Certificate>?, authType: String?) {
-                        TODO("not implemented")
-                    }
-
-                    override fun getAcceptedIssuers(): Array<X509Certificate> {
-                        TODO("not implemented")
-                    }
-
-                }),
+                trustManagers,
```
Comment (ateska) [inline seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt:106]: call to `createTrustManagers()`

File: seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt [hunk 4]
```diff
@@ -36,6 +38,9 @@ class SeaCat(
     val identity = Identity(this)
     val peers = PeerProvider(this)
 
+    /** Array of trust managers */
+    val trustManagers : Array<TrustManager>
+
```
Comment (ateska) [inline seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt:43]: No member variable needed, I guess.

**Review by ateska — CHANGES_REQUESTED** [review]
See in code.

File: seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt (hunk 1 above)
Comment (mpavelka) [reply seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt:19]: This is needed in case we want to allow overriding of `createTrustManagers()`
Do we?

Comment (ateska) [reply seacat/src/main/java/com/teskalabs/seacat/SeaCat.kt:19]: I would like to introduce an alternative way of this (e.g. SeaCat controller).

**Pull request merged**